### PR TITLE
Update AppService to support custom entrypoint for dynamic configuration

### PR DESCRIPTION
**Pull request by wbreza**
Updates app service todo templates to run custom entrypoint script that generates `env-config.js` used by static web application.

### CODE REVIEW

File: templates/todo/web/react-fluent/package.json [hunk 1]
```diff
@@ -20,8 +20,8 @@
     "build": "tsc -b .",
     "pretest": "npm run envconfig",
     "test": "if-node-version '>= 17' && react-scripts --openssl-legacy-provider test || react-scripts test",
-    "prepackage": "npm run envconfig",
     "package": "if-node-version '>= 17' && react-scripts --openssl-legacy-provider build || react-scripts build",
+    "postpackage": "cp ./entrypoint.sh ./build",
```
Comment (wbreza) [inline templates/todo/web/react-fluent/package.json:24]: This ensures our custom `entrypoint.sh` is available on the webserver for custom command.

File: templates/todo/web/react-fluent/package.json [hunk 2]
```diff
@@ -20,8 +20,8 @@
     "build": "tsc -b .",
     "pretest": "npm run envconfig",
     "test": "if-node-version '>= 17' && react-scripts --openssl-legacy-provider test || react-scripts test",
-    "prepackage": "npm run envconfig",
```
Comment (wbreza) [inline templates/todo/web/react-fluent/package.json:23]: We no longer need to gen env config file on package.

File: templates/todo/web/react-fluent/package.json (hunk 1 above)
Comment (weikanglim) [reply templates/todo/web/react-fluent/package.json:24]: Makes sense. There isn't a different way to declare this as an `asset` file, i'm guessing?

File: templates/todo/web/react-fluent/entrypoint.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (weikanglim) [inline templates/todo/web/react-fluent/entrypoint.sh:1]: The Dockerfile we use has `bash`, yeah?

Comment (wbreza) [reply templates/todo/web/react-fluent/entrypoint.sh:1]: I will double check.  I was getting some errors that we're using bash-isms here.

Comment (jongio) [reply templates/todo/web/react-fluent/entrypoint.sh:1]: Ask chatgpt to convert it for you.

**Review by vhvb1989 — APPROVED** [review]

**Review by weikanglim — APPROVED** [review]

**Comment by pamelafox** [comment]
We upgraded to use this appservice.bicep in the openai sample a few weeks ago, and I've noticed numerous developers experiencing an issue where their App Service app doesn't actually have all the settings (it has none!). I'm wondering it it's related to this change, making the settings a separate module. I don't know for sure since it doesn't happen 100% of the time, but the timing and change seems suspicious. Why did the settings have to become a separate module?

**Pull request merged**